### PR TITLE
added stuff

### PR DESCRIPTION
**Pull request by Dex-theDev**
Lotsa stuff, will be adding comments to specific files

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -31,6 +31,12 @@
 				<groupId>org.springframework.boot</groupId>
 				<artifactId>spring-boot-starter-web</artifactId>
 			</dependency>
+		<dependency>
+  <groupId>io.norberg</groupId>
+  <artifactId>auto-matter</artifactId>
+  <version>0.26.1</version>
+  <scope>provided</scope>
+</dependency>
 	</dependencies>
```
Comment (Dex-theDev) [inline pom.xml:40]: So i found out about @Automatter at Spotify, basically it allows your models/entities to act as interfaces with a bunch of built in stuff so we don't have to waste time creating getters/setters, toString(), amongst a ton of other things. It also adds a builder class so we can dynamically build our objects pretty easily.

File: pom.xml [hunk 2]
```diff
@@ -39,6 +45,27 @@
...
+    </configuration>
+    <executions>
+        <execution>
+            <goals>
+                <goal>check</goal>
+            </goals>
+        </execution>
+    </executions>
+</plugin>
 		</plugins>
```
Comment (Dex-theDev) [inline pom.xml:69]: added a maven formatting plugin to google format my code so it looks professhh!

File: src/main/java/com/healthapp/project/Entity/Medication.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.healthapp.project.Entity;
+
+import io.norberg.automatter.AutoMatter;
+
+@AutoMatter
+public interface Medication {
+    int id();
+    String medication_name();
+    String strength();
+}
```
Comment (Dex-theDev) [inline src/main/java/com/healthapp/project/Entity/Medication.java:10]: So with a few lines, there's so many convenient things built in here. It also handles JSON mapping for me so i dont have to worry about that as well. I'll probably circle back and do the same to the User entity once i handle the medication api

**Review by rosannaNaomy — APPROVED** [review]
This looks great Dex! I really like the detailed comments you left on some lines of code. It makes it easier to understand and I'm glad to know why you're using the plugin. I see how you've separated classes into packages, like DAO, Controller and Service. Are you setting up some type of architecture here?

**Comment by Dex-theDev** [comment]
Lol i'm aiming to, that way when you want to contribute, you know exactly where to go!

**Pull request merged**